langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchOrganicResult.java

```diff
@@ -0,0 +1,204 @@
+package dev.langchain4j.web.search;
+
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.segment.TextSegment;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+
+/**
+ * Represents an organic search results are the web pages that are returned by the search engine in response to a search query.
+ * This includes the title, link, content, and metadata of the web page.
+ * <p>
+ * These results are typically ranked by relevance to the search query.
+ * <p>
+ */
+public class WebSearchOrganicResult {
+
+    private final String title;
+    private final String link;
+    private final String content;
```

Comment:
nitpick: "content" sounds like it is a content of a web page, which is not the case, right?
Should it be named "snippet" or something similar?

Reply:
yes, initially it was named snipped and I think that because of the confusion that you thought it returned the full content of the page, you asked me to change it to content . Sorry I should have explained it better to you from the beginning. I'll return it as it was.

Reply:
No worries! Sorry for misunderstanding! Most important is that we are on the same page now :)